src/devana/utility/init_params.py

```diff
@@ -0,0 +1,44 @@
+from typing import Callable, Any, Optional, Set
+from inspect import signature, BoundArguments
+from functools import wraps
+
+
+def init_params(skip: Optional[Set[str]] = None) -> Callable:
+    """A decorator that assigns method parameters as instance attributes.
+    The attribute must have a setter or exist as an instance or class variable."""
+
+    def decorator(_classmethod: Callable) -> Callable:
+
+        def has_setter(instance: object, name: str) -> bool:
+            """Checks if the class attribute is a property with a defined setter method."""
+            try:
+                maybe_property = getattr(instance.__class__, name)
+            except AttributeError:
+                return False
+            return isinstance(maybe_property, property) and maybe_property.fset is not None
+
+        def has_attr(instance: object, name: str) -> bool:
+            """Checks if the attribute is directly in the instance or if it's a class attribute,
+            excluding properties."""
+            if hasattr(instance.__class__, name):
+                if not isinstance(getattr(instance.__class__, name, property()), property):
```

Comment:
You check if an attribute exists, adding a default one in this case is a bit pointless.